docs/reference/builder.md

```diff
@@ -1291,6 +1295,90 @@ The `STOPSIGNAL` instruction sets the system call signal that will be sent to th
+Windows is `cmd /S /C`. 
+
+The `SHELL` instruction  is particularly useful on Windows where there are
+two commonly used and quite different native shells: `cmd` and `powershell`, as
+well as alternate shells available including `sh`.
+
+The `SHELL` instruction can make writing Dockerfiles simpler when the *shell* form
+is used in commands such as `RUN`. The following example is a common pattern
+found on Windows which can be streamlined by using the `SHELL` instruction: 
+
+    ...
+    RUN powershell -command Execute-MyCmdlet -param1 "Value"
+    ... 
+
+The command invoked by docker will be:
+
+    cmd /S /C powershell -command Execute-MyCmdlet -param1 "Value"
+   
+ This is inefficient for two reasons. First, there is an un-necessary cmd.exe command
+ processor (aka shell) being invoked. Second, each `RUN` instruction in the *shell*
+ form requires an extra `powershell -command` prefixing the command.
+ 
+ In this example, both efficiencies are removed, and the dockerfile contains more
+ natural syntax for PowerShell users on Windows:
+  
+    FROM windowsservercore
+    SHELL powershell -command
+    RUN New-Item -ItemType Directory C:\\Example
+    WORKDIR C:\\Example
+    ADD Execute-MyCmdlet.ps1 .
+    RUN .\Execute-MyCmdlet -sample 'hello world'
+
+Resulting in:
+
+    PS E:\docker\build\shell> docker build -t test .
+    Sending build context to Docker daemon 3.072 kB
+    Step 1 : FROM windowsservercore
+     ---> dbfee88ee9fd
+    Step 2 : SHELL powershell -command
+     ---> Running in 40e288a10189
+     ---> ccda3165f813
+    Removing intermediate container 40e288a10189
+    Step 3 : RUN New-Item -ItemType Directory C:\\Example
+     ---> Running in 1597e77c3419
+    
+    
+        Directory: C:\
+    
+    
+    Mode                LastWriteTime         Length Name
+    ----                -------------         ------ ----
+    d-----         5/3/2016   4:12 PM                Example
+    
+    
+     ---> a6b565fcbf93
+    Removing intermediate container 1597e77c3419
+    Step 4 : WORKDIR C:\\Example
+     ---> Running in 601fcb646ad9
+     ---> c1da01922b95
+    Removing intermediate container 601fcb646ad9
+    Step 5 : ADD Execute-MyCmdlet.ps1 .
+     ---> 018d734706b1
+    Removing intermediate container bbab8f5d69bb
+    Step 6 : RUN .\Execute-MyCmdlet -sample 'hello world'
+     ---> Running in 7ef63fe3d2b3
+    Hello from Execute-MyCmdlet.ps1 - passed hello world
+     ---> 403cdaab2082
+    Removing intermediate container 7ef63fe3d2b3
+    Successfully built 403cdaab2082
+    PS E:\docker\build\shell>
+
+The `SHELL` instruction could also be used to modify the way in which
+a shell operates. For example, using `SHELL cmd /S /C /V:ON|OFF` on Windows, delayed
+environment variable expansion semantics could be modified.
+    
+The `SHELL` instruction can also be used on Linux should an alternate shell be
+required such `zsh`, `csh`, `tcsh` and others.
+
```

Comment:
let's add something about how it can appear more than once and then influences all subsequent RUN, CMD and ENTRYPOINT cmds.